var/spack/repos/builtin/packages/gobject-introspection/package.py

```diff
@@ -56,6 +56,10 @@ class GobjectIntrospection(MesonPackage, AutotoolsPackage):
     # https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/283
     depends_on("libffi@:3.3", when="@:1.72")  # libffi 3.4 caused seg faults
     depends_on("python")
+    with when("^python@3.12:"):
+        depends_on("py-setuptools")
```

Comment:
```suggestion
        depends_on("py-setuptools@48:")
```
ref: https://github.com/pypa/setuptools/commit/bb9fb1fcfe37c1ef1e29e1e6d1fc4e483c743380